integration/raft/channel_participation_test.go

```diff
@@ -124,43 +125,89 @@ var _ = Describe("ChannelParticipation", func() {
 
 			for _, o := range orderers {
 				By("joining " + o.Name + " to channel as a member")
-				channelparticipation.Join(network, o, "participation-trophy", genesisBlock, expectedChannelInfo)
-				channelparticipation.ListOne(network, o, expectedChannelInfo)
+				channelparticipation.Join(network, o, "participation-trophy", genesisBlock, expectedChannelInfoPT)
+				channelparticipation.ListOne(network, o, expectedChannelInfoPT)
 			}
 
 			By("waiting for the leader to be ready")
-			findLeader(ordererRunners)
+			leader := findLeader(ordererRunners)
 
 			By("ensuring the channel is usable by submitting a transaction to each orderer")
 			env := CreateBroadcastEnvelope(network, peer, "participation-trophy", []byte("hello"))
-			expectedBlockNum := 1
 			for _, o := range orderers {
+				By("submitting transaction to " + o.Name)
+				expectedChannelInfoPT.Height++
 				resp, err := ordererclient.Broadcast(network, o, env)
 				Expect(err).NotTo(HaveOccurred())
 				Expect(resp.Status).To(Equal(common.Status_SUCCESS))
-				expectedChannelHeight := map[string]int{"participation-trophy": expectedBlockNum}
-				assertBlockReception(expectedChannelHeight, orderers, peer, network)
+				expectedBlockNumPerChannel := map[string]int{"participation-trophy": int(expectedChannelInfoPT.Height - 1)}
+				assertBlockReception(expectedBlockNumPerChannel, orderers, peer, network)
 
 				By("checking the channel height")
-				expectedChannelInfo.Height = uint64(expectedBlockNum + 1)
-				channelparticipation.ListOne(network, o, expectedChannelInfo)
-				expectedBlockNum++
+				channelparticipation.ListOne(network, o, expectedChannelInfoPT)
 			}
 
 			By("joining orderer1 to another channel as a member")
 			genesisBlock2 := applicationChannelGenesisBlock(network, []*nwo.Orderer{orderer1}, peer, "another-participation-trophy")
-			expectedChannelInfo = channelparticipation.ChannelInfo{
+			expectedChannelInfoAPT := channelparticipation.ChannelInfo{
 				Name:            "another-participation-trophy",
 				URL:             "/participation/v1/channels/another-participation-trophy",
 				Status:          "active",
 				ClusterRelation: "member",
 				Height:          1,
 			}
-			channelparticipation.Join(network, orderer1, "another-participation-trophy", genesisBlock2, expectedChannelInfo)
-			channelparticipation.ListOne(network, orderer1, expectedChannelInfo)
+			channelparticipation.Join(network, orderer1, "another-participation-trophy", genesisBlock2, expectedChannelInfoAPT)
+			channelparticipation.ListOne(network, orderer1, expectedChannelInfoAPT)
 
-			By("listing all channels")
+			By("listing all channels for orderer1")
 			channelparticipation.List(network, orderer1, []string{"participation-trophy", "another-participation-trophy"})
+
+			By("removing orderer1 from the consenter set")
+			channelConfig := nwo.GetConfig(network, peer, orderer2, "participation-trophy")
```

Comment:
This is indeed the recommended flow. First remove from the consenters set, then remove from orderer. However - I have two comments:
1. The reverse flow should be working as well, you just have to wait for the remaining consenters to re-elect a leader in case the leader was the one removed.
2. When a raft chain detects eviction, it should turn into a follower. This is not yet implemented, the chain will just print something and exit. After https://jira.hyperledger.org/browse/FAB-17915 that will change.  So when that happens one should also check that the orderer behaves as expected when evicted from a channel.

I would insert a TODO and/or a Jira in the integration test story to catch these two test cases and cover then later.

Reply:
Sounds good. I'lll open sub-tasks in the integration test task.